app/samples/sample-browsemap-details-window/src/main/java/com/sygic/samples/detail/CustomDetailsViewFactory.kt

```diff
@@ -1,54 +1,44 @@
 package com.sygic.samples.detail
 
-import android.os.Parcel
-import android.os.Parcelable
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import android.widget.TextView
 import com.sygic.modules.common.detail.DetailsViewFactory
 import com.sygic.samples.R
-import com.sygic.ui.common.sdk.data.PoiData
+import com.sygic.sdk.map.`object`.data.ViewObjectData
+import com.sygic.ui.common.sdk.data.BasicData
+import com.sygic.ui.common.sdk.extension.getFormattedLocation
+import kotlinx.android.parcel.Parcelize
 
-class CustomDetailsViewFactory() : DetailsViewFactory() {
+@Parcelize
+class CustomDetailsViewFactory : DetailsViewFactory() {
 
-    override fun getDetailsView(inflater: LayoutInflater, container: ViewGroup, poiData: PoiData): View {
+    override fun getDetailsView(inflater: LayoutInflater, container: ViewGroup, data: ViewObjectData): View {
         val root = inflater.inflate(R.layout.layout_info_window, container, false)
 
-        //fill layout with data from marker's PoiData
-        val addressComponent = poiData.getAddressComponent()
-        addressComponent.formattedTitle.let {
-            if (it.isEmpty()) {
-                root.findViewById<TextView>(R.id.title).visibility = View.GONE
-            } else {
-                root.findViewById<TextView>(R.id.title).text = it
-            }
-        }
-        addressComponent.formattedSubtitle.let {
-            if (it.isEmpty()) {
-                root.findViewById<TextView>(R.id.snippet).visibility = View.GONE
-            } else {
-                root.findViewById<TextView>(R.id.snippet).text = it
+        // fill layout with data from marker's Data
+        data.payload.let { payload ->
```

Comment:
tu je to tiez take, ze tu tie lety tu vytvaraju zanaranie a zbytocne lambdy... citatelne je to v pohode, ale ze by to bol taky prinos neviem

Reply:
je to ten stejný případ, jako jsem popisoval výše.